### PR TITLE
Avoid NPE in ant module suite's "Project Properties" dialog.

### PR DESCRIPTION
**Pull request by errael**
`ProjectNav > some-module-suite > properties > Libraries`
when `<module-suite-dir>/nbproject/platform.properties`
has `cluster.path=... <maven-nbm-dir>/target/nbm/clusters/extra`

This fix prevents a hung dialog (cancel terminates it).
It also makes it simpler to incorporate a maven based
nbm into an existing module suite as an external cluster.

Note that specifying a maven nbm project in cluster.path currently works for running the suite. It's just this dialog that hangs.

### CODE REVIEW

File: apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ClusterUtils.java [hunk 1]
```diff
@@ -169,8 +169,11 @@ public static Set<ClusterInfo> evaluateClusterPath(File root, PropertyEvaluator
                     prj = _prj;
                 } else {
                     NbModuleProvider prov2 = _prj.getLookup().lookup(NbModuleProvider.class);
-                    if (prov2 != null && /* XXX is there a better way? */ cd.equals(prov2.getModuleJarLocation().getParentFile().getParentFile())) {
-                        prj = _prj;
+                    if (prov2 != null) {
+                        File fi = prov2.getModuleJarLocation();
+                        if (fi != null && /* XXX is there a better way? */ cd.equals(fi.getParentFile().getParentFile())) {
```
Comment (mbien) [inline apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ClusterUtils.java:174]: `getParentFile().getParentFile()`
sounds like a second potential NPE ;)

Comment (errael) [reply apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ClusterUtils.java:174]: Technically yes; but no, not really. The contract is that a returned File is somewhere inside a project, it's parent can't be null.

For example, elsewhere in `ClusterUtils` there's
```
            File jar = nbmp.getModuleJarLocation();
            if (jar != null) {
                return jar.getParentFile().getParentFile();
            }
```
There are other examples scattered through other files.

Comment (errael) [reply apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ClusterUtils.java:174]: I hadn't seen this before
```
    /**
     * Returns location of built module JAR (file need not to exist).
     *
     * Currently (6.7) used only for suite-chaining. May return <tt>null</tt>,
     * module project cannot be chained into another suite in such case.
     * @return location of built module JAR
     */
    @CheckForNull File getModuleJarLocation();
```

Comment (mbien) [reply apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ClusterUtils.java:174]: could be put into a utility method `getRootDir(NbModuleProvider prov)`. Would be self explanatory and could do the null check there.

Comment (errael) [reply apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ClusterUtils.java:174]: > could ... getRootDir(NbModuleProvider prov)

Something to consider for a rework of this ancient code, not holding my breath :-) Seems out of scope for fixing a missing null check.

Comment (mbien) [reply apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ClusterUtils.java:174]: if you want you can add another commit. first fixes the bug, second extracts the two occurrences into a utility method. This keeps bugfix and code improvement separate.

There is no need to wait with small cleanups like this.

**Comment by errael** [comment]
Turns out `ClusterUtils` already has `getClusterDirectory(Project prj)`. The only difference is whether suite or standalone is checked first.

**Review by mbien — APPROVED** [review]
awesome, looks good to me.

**Pull request merged**